broker/broker.py

```diff
@@ -82,6 +82,7 @@ class Broker:
     def __init__(self, **kwargs):
         kwargs = helpers.resolve_file_args(kwargs)
         logger.debug(f"Broker instantiated with {kwargs=}")
+        self.skip_checkin =  kwargs.pop("skip_checkin", False)
```

Comment:
If this is going to only affect checkins from the context manager, then I think the argument needs to be a bit more clear for that. Otherwise, setting this on class instantiation could be interpreted to mean that this Broker instance won't checkin anything.
```suggestion
        self.skip_context_checkin =  kwargs.pop("skip_context_checkin", False)
```